sdk/cosmos/azure-spring-data-cosmos-test/src/test/java/com/azure/spring/data/cosmos/ContainerLockIT.java

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.spring.data.cosmos;
+
+import com.azure.spring.data.cosmos.core.CosmosTemplate;
+import com.azure.spring.data.cosmos.repository.TestRepositoryConfig;
+import com.azure.spring.data.cosmos.repository.support.CosmosEntityInformation;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import java.time.Duration;
+import java.time.OffsetDateTime;
+
+import static org.junit.Assert.assertTrue;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = TestRepositoryConfig.class)
+public class ContainerLockIT {
+
+    private static final Duration SHORT_LEASE_DURATION = Duration.ofSeconds(3);
+
+    @Autowired
+    private CosmosTemplate template;
+    private static CosmosTemplate staticTemplate;
+
+    private ContainerLock lock;
+    private ContainerLock otherLock;
+
+    @Before
+    public void setup() {
+        staticTemplate = template;
+        CosmosEntityInformation entityInfo = new CosmosEntityInformation(ContainerLock.LockEntry.class);
+        template.createContainerIfNotExists(entityInfo);
+        lock = new ContainerLock(template, entityInfo.getContainerName(), SHORT_LEASE_DURATION);
+        otherLock = new ContainerLock(template, entityInfo.getContainerName(), SHORT_LEASE_DURATION);
+    }
+
+    @After
+    public void cleanup() {
+        releaseLockIgnoreException(lock);
+        releaseLockIgnoreException(otherLock);
+    }
+
+    private void releaseLockIgnoreException(ContainerLock lock) {
+        try {
+            lock.release();
+        } catch (Exception ex) {
+            // ignore
+        }
+    }
+
+    @AfterClass
+    public static void afterClassCleanup() {
+        staticTemplate.deleteContainer(new CosmosEntityInformation<>(ContainerLock.LockEntry.class).getContainerName());
```

Comment:
in the `@After` API above, container will be automatically deleted, no ?
If so, whey do we need the `afterClassCleanup` ?

Reply:
`@After` release the lock and releasing the lock just deletes the lock document, it does not delete the container.

Reply:
Sounds good.